src/Workspaces/Core/Portable/SymbolKey/SymbolKey.SymbolKeyReader.cs

```diff
@@ -566,7 +566,8 @@ protected override string CreateResultForString(int start, int end, bool hasEmbe
                     var syntaxTree = GetSyntaxTree(filePath);
                     if (syntaxTree == null)
                     {
-                        failureReason = $"({nameof(ReadLocation)} failed -> '{filePath}' not in compilation)";
+                        var filePaths = string.Join(", ", this.Compilation.SyntaxTrees.Select(t => $"'{t.FilePath}'"));
+                        failureReason = $"({nameof(ReadLocation)} failed -> '{filePath}' not in compilation filepaths: {filePaths})";
```

Comment:
this will help us know things like:

1. is the compilation just missing syntax trees altogether.
2. does it include files, but is missing the oen file we're looking for.
3. does it include the file, but maybe doesn't match the filepath based on casing.
3. does it include the file, but maybe doesn't match teh file on some other normalization.

Reply:
This is going to be a huge string, but as long as this is only for exceptional circumstances, it should be fine.

Reply:
This is getting reported as NFE right? Probably will get truncated (I'm not 100% on this). 

Can we have a separate failure message for 1 above, and 2,3,4 can be grouped by searching for _similar_ files in the trees?